### PR TITLE
Testing

### PR DESCRIPTION
**Pull request by Tagadda**
## Major changes

- Convert to packaging v2 (#383)
- Use SMTP instead of sendmail, thanks to the `allow_mail` option for `system_user`
- Use logrotate ([97e74bc](https://github.com/YunoHost-Apps/mastodon_ynh/pull/394/commits/97e74bc498320f690f94b81b2911aa0da5bd846a))

## PR Status

- [x] Code finished and ready to be reviewed/tested
- [x] The fix/enhancement were manually tested (if applicable)

## Automatic tests

Automatic tests can be triggered on https://ci-apps-dev.yunohost.org/ *after creating the PR*, by commenting "!testme", "!gogogadgetoci" or "By the power of systemd, I invoke The Great App CI to test this Pull Request!". (N.B. : for this to work you need to be a member of the Yunohost-Apps organization)

### CODE REVIEW

File: doc/PRE_INSTALL.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
+## Important points to read before installing
+
+- **Mastodon** require a dedicated **root domain**, eg. `domaine.tld` or `mastodon.domain.tld`, with no other apps installed on that domain. You can't change the domain once installed.
+- The user choosen during the installation is automatically created in Mastodon with admin rights
+- It seems important to close the inscriptions for your Mastodon, so that it remains a private body. We invite you to block remote malicious instances from the administration interface. You can also add text on your home page.
```
Comment (alexAubin) [inline doc/PRE_INSTALL.md:5]:
```suggestion
- It seems important to close registrations for your Mastodon, so that it remains a private body. We invite you to block remote malicious instances from the administration interface. You can also add text on your home page.
```

File: scripts/install [hunk 2]
```diff
@@ -159,122 +67,91 @@ ynh_app_setting_set --app="$app" --key=secret_key_base --value="$secret_key_base
...
 	ynh_exec_as $app $ynh_ruby_load_path $ld_preload bin/bundle config deployment 'true'
 	ynh_exec_as $app $ynh_ruby_load_path $ld_preload bin/bundle config without 'development test'
 	ynh_exec_as $app $ynh_ruby_load_path $ld_preload bin/bundle config set force_ruby_platform true
 	ynh_exec_as $app $ynh_ruby_load_path $ld_preload bin/bundle install -j$(getconf _NPROCESSORS_ONLN)
+	# Building assets
 	ynh_use_nodejs
 	ynh_exec_warn_less ynh_exec_as $app $ynh_node_load_PATH yarn install --pure-lockfile
 	echo "SAFETY_ASSURED=1">> $config
-	ynh_exec_warn_less ynh_exec_as $app RAILS_ENV=production $ynh_ruby_load_path $ld_preload bin/bundle exec rails db:setup --quiet
+	ynh_exec_warn_less ynh_exec_as $app RAILS_ENV=production $ynh_ruby_load_path $ld_preload bin/bundle exec rails db:migrate --quiet
```
Comment (alexAubin) [inline scripts/install:104]: We sure about this ? This seems unrelated to v2 ? (Maybe this is legit idk)

**Review by alexAubin — APPROVED** [review]
LGTM, super minor stuff

File: scripts/install (hunk 2 above)
Comment (Tagadda) [reply scripts/install:104]: Yup, did this on purpose because with `db:setup`, rake try to create the DB and fails (and with packaging v2 we don't have the permission to do so? I don't actually know why this doesn't fail with v1...)

File: doc/ADMIN.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
+## Administrate with tooctl
+
+`$ (cd /var/www/mastodon/live && sudo -u mastodon RAILS_ENV=production PATH=/opt/rbenv/versions/mastodon/bin bin/tootctl --help)`
```
Comment (tituspijean) [inline doc/ADMIN.md:3]: Maybe we can take advantage of `yunohost app shell mastodon` ;)

Comment (Tagadda) [reply doc/ADMIN.md:3]: Since there's multiple systemd services, we need to set a `service` setting in install and upgrade scripts, right ?

**Comment by Tagadda** [comment]
!testme
